### DISCUSSION TITLE
URL Host agnostic cache key

### DISCUSSION THREAD

**Discussion by yringler** [SixLabors/ImageSharp.Web#145] [Ideas] [answered]
We host a lot of (nonprofit) sites on different domains (from the save servers), and want all of them to cache images to the same place, even though the domain is different.
There is at least one workaround.
~~The first relies on an implementation detail~~ https://github.com/SixLabors/ImageSharp.Web/blob/451da152293154bff58c6943b37ca5b26ca9f26b/src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs#L497
~~I suspect there are other ways to get the host / full URL with host? If there are, it's a brittle work around.~~
The second is better? But it disturbs my aesthetic sense to have ImageSharp carefully create a URI string, just for me to parse it and clumsily remove the host.

Do you think that there should be something added to more gracefully support this?
Maybe add another method to the interface which takes an HttpRequest instead of a string? It could have a default implementation which creates the string as in the line quoted above.

Or a new `ImageSharpMiddlewareOptions.IgnoreHost` option?

With middleware:
```csharp
app.Use(async (context, next) =>
{
  context.Request.Host = new Microsoft.AspNetCore.Http.HostString("ourmainhost.org");
  await next();
});
app.UseImageSharp();
app.UseStaticFiles();
```

And by making a custom ICacheHash
**EDIT**: This doesn't work, because the string isn't a full URI
```csharp
using Microsoft.Extensions.Options;
using SixLabors.ImageSharp.Web.Caching;
using SixLabors.ImageSharp.Web.Middleware;
using System;

namespace ImageProcessor
{
	public class HostAgnosticCacheHash : ICacheHash
	{
		private readonly CacheHash cacheHash;

		public HostAgnosticCacheHash(IOptions<ImageSharpMiddlewareOptions> options)
		{
			cacheHash = new CacheHash(options);
		}

		public string Create(string value, uint length)
		{
			var sanatizedUri = new Uri(value);
			
			return cacheHash.Create(value.Replace(sanatizedUri.Host, ""), length);
		}
	}
}

```

**Comment by JimBobSquarePants** [comment, marked as answer]
>ImageSharpMiddlewareOptions.IgnoreHost

That's the best option. Happy to help if you want to open a PR.

**Reply by yringler** [reply]
Ok, great! I'll open a PR shortly

**Reply by JimBobSquarePants** [reply]
Awesome!

**Reply by yringler** [reply]
Opened https://github.com/SixLabors/ImageSharp.Web/pull/147
